File: binarySearch.java
import java.util.Scanner;

  public static void binarySearch(int arr[], int first, int last, int key)
  {  
   int mid = (first + last)/2;  
   while( first <= last ){  
      if ( arr[mid] < key ){  
        first = mid + 1;     
      }else if ( arr[mid] == key ){  
        System.out.println("Element is found at index: " + mid);  
        break;  
      }else{  
         last = mid - 1;  
      }  
      mid = (first + last)/2;  
   }  
   if ( first > last ){  
      System.out.println("Element is not found!");  
   }  
 }  

 public static void main(String args[])
 {  
   Scanner s=new Scanner(System.in);
   int n=s.nextInt() ;
   int[] arr=new int[n];
   
   for(int i=0;i<arr.length;i++)
   {
     arr[i]=s.nextInt();
   }
   int key=s.nextInt();
   int last=arr.length-1;
   int result=binarySearch(arr,0,last,key);
 }

File: Binary_search.py
def binary_search(x,key):
  s=0
  end=(len(lst)-1)
  while(s<=end):
    mid=(s+end)//2
  if(lst[mid]<key):
    start=mid+1
  elif(lst[mid]>key):
    end=mid-1
  else:
    return mid
  return -1

lst=[]
n=1
while(n!=0):
  n=int(input("please enter the element:"))
  if(n!=0):
    lst.append(n)
print(lst)

m=int(input("please enter the key:"))

y=binary_search(lst,m)
print(y)
